### PR TITLE
Kafka extra producer configuration enhancement

### PR DESCRIPTION
**Pull request by dmsolr**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaProducerManager.java [hunk 1]
```diff
@@ -102,6 +104,10 @@ public void run() {
         Properties properties = new Properties();
         properties.setProperty(
             ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, KafkaReporterPluginConfig.Plugin.Kafka.BOOTSTRAP_SERVERS);
+
+        Map<String, String> producerConfig = (Map<String, String>) new Gson()
```
Comment (kezhenxu94) [inline apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaProducerManager.java:108]: Extract `new Gson()` as a field

**Review by kezhenxu94 — COMMENTED** [review]
Documentation and `agent.conf` should be updated, and indicate that if a property is set in both `PRODUCER_CONFIG` and `PRODUCER_CONFIG_JSON`, the later overrides the former

File: docs/en/setup/service-agent/java-agent/How-to-enable-kafka-reporter.md [hunk 2]
```diff
@@ -12,8 +12,10 @@ collector.backend_service=${SW_AGENT_COLLECTOR_BACKEND_SERVICES:127.0.0.1:11800}
 
 # Kafka producer configuration
 plugin.kafka.bootstrap_servers=${SW_KAFKA_BOOTSTRAP_SERVERS:localhost:9092}
-plugin.kafka.producer_config[delivery.timeout.ms]=12000
 plugin.kafka.get_topic_timeout=${SW_GET_TOPIC_TIMEOUT:10}
+
+# Configure extra Kafka Producer configuration
+plugin.kafka.producer_config[delivery.timeout.ms]=12000
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/How-to-enable-kafka-reporter.md:18]: I think comments in the `KafkaReporterPluginConfig` should be a separate section in this doc. 
It is better to remove `plugin.kafka.producer_config[delivery.timeout.ms]=12000` from here.

Adding `Advanced Kafka Configs` maybe better

File: docs/en/setup/service-agent/java-agent/How-to-enable-kafka-reporter.md [hunk 3]
```diff
@@ -12,8 +12,10 @@ collector.backend_service=${SW_AGENT_COLLECTOR_BACKEND_SERVICES:127.0.0.1:11800}
 
 # Kafka producer configuration
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/How-to-enable-kafka-reporter.md:13]: One missing change

> In other words, the following configure cannot be omitted.

These words should be updated. All configurations have been activated in the `agent.config`.

File: apm-sniffer/config/agent.config [hunk 4]
```diff
@@ -216,12 +216,12 @@ plugin.dubbo.consumer_provider_length_threshold=${SW_PLUGIN_DUBBO_CONSUMER_PROVI
...
 plugin.kafka.get_topic_timeout=${SW_GET_TOPIC_TIMEOUT:10}
-#  Kafka producer configuration.
-plugin.kafka.consumer_config=${SW_PLUGIN_KAFKA_CONSUMER_CONFIG:}
 # Kafka producer configuration. Read [producer configure](http://kafka.apache.org/24/documentation.html#producerconfigs)
 # to get more details. Check [Kafka report doc](How-to-enable-kafka-reporter.md) for more details and examples.
-plugin.kafka.producer_config=${SW_PLUGIN_KAFKA_PRODUCER_CONFIG:}
-#  Specify which Kafka topic name for Meter System data to report to. 
+plugin.kafka.producer_config=${sw_plugin_kafka_producer_config:}
+# Configure Kafka Producer configuration in JSON format. Notice it will be overridden by plugin.kafka.producer_config[key], if the key duplication.
+plugin.kafka.producer_config_json=${sw_plugin_kafka_producer_config_json:}
```
Comment (wu-sheng) [inline apm-sniffer/config/agent.config:223]:
```suggestion
plugin.kafka.producer_config_json=${SW_PLUGIN_KAFKA_PRODUCER_CONFIG_JSON:}
```

File: docs/en/setup/service-agent/java-agent/README.md [hunk 5]
```diff
@@ -153,8 +153,8 @@ property key | Description | Default |
 `plugin.dubbo.provider_arguments_length_threshold`| When `plugin.dubbo.collect_provider_arguments` is `true`, Arguments of length from the front will to the OAP backend |`256`| 
 `plugin.kafka.bootstrap_servers`| A list of host/port pairs to use for establishing the initial connection to the Kafka cluster. | `localhost:9092`|
 `plugin.kafka.get_topic_timeout`| Timeout period of reading topics from the Kafka server, the unit is second. |`10`|
-`plugin.kafka.consumer_config`| Kafka producer configuration. ||
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/README.md:156]: I think this still works, right?

**Comment by wu-sheng** [comment]
@kezhenxu94 I think we should add a system environment variable for the agent setup, as you update `agent.config`.
And we should cut the current `readme.md` into different documents, the page is too long to read.

**Comment by kezhenxu94** [comment]
> @kezhenxu94 I think we should add a system environment variable for the agent setup, as you update `agent.config`.

Yes, it's better to add a column for environment variable name in the agent setup doc

**Comment by wu-sheng** [comment]
> And we should cut the current readme.md into different documents, the page is too long to read.

Create #7532 for further update.

File: docs/en/setup/service-agent/java-agent/README.md (hunk 5 above)
Comment (dmsolr) [reply docs/en/setup/service-agent/java-agent/README.md:156]: Not, just forget to remove it from docs. This was originally reserved for `profile task` communication with OAP. But we discard this way. Agent communicate with OAP one way.

Comment (dmsolr) [reply docs/en/setup/service-agent/java-agent/README.md:156]: Not, just forget to remove it from docs. This was originally reserved for `profile task` communication with OAP. But we discard this way. Agent communicate with OAP one way.

**Review by wu-sheng — APPROVED** [review]

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Pull request merged**